src/daft-plan/src/physical_plan.rs

```diff
@@ -206,6 +211,8 @@ impl PhysicalPlan {
             Self::IcebergWrite(..) => {
                 ClusteringSpec::Unknown(UnknownClusteringConfig::new(1)).into()
```

Comment:
why daft is giving 1 num_partitions in iceberg case  ?.

Reply:
Actually this might be an artifact of when I was performing the iceberg table transaction in the execution plan rather  than in the dataframe method body. 
However when I tested in Ray, I found out that the IcebergTable object was not pickle-able. So I had to move it out to `write_iceberg`. 

This means though the number of partitions should probably be equal to input partitions since each one is just returning the `DataFile`.

Reply:
> However when I tested in Ray, I found out that the IcebergTable object was not pickle-able. So I had to move it out to write_iceberg.

maybe it can also resolve my ray test error `serialization: cannot pickle 'deltalake._internal.RawDeltaTable' object` .
but will it have any negative effect on write performance and partitioning data . (setting up ray is hard 🧙‍♂️ )

Reply:
hey @samster25 i need help with debuging ray test error `!!! FAIL serialization: cannot pickle 'deltalake._internal.RawDeltaTable' object
WARNING: Did not find non-serializable object in [WriteDeltaLake(base_path='/tmp/pytest-of-runner/pytest-0/`
I naively thought 
`ClusteringSpec::Unknown(UnknownClusteringConfig::new(1)).into()` would be a quick, easy fix.

Reply:
@sherlockbeard sorry for the delay! will spend some time today looking into this :)